src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.py

```diff
@@ -43,6 +46,88 @@ def target_fn(*args, **kwargs):
         raise e
 
 
+class TestMultiprocessing(unittest.TestCase):
+    mp = MultiProcessing()
+
+    def test_multiprocessing_spawn(self):
+        """
+        Spawns an actor.
+        Checks that an actor is spawned successfully.
+        """
+        self.mp.start()
+        builder = Builder()
+
+        # Build 5 actors
+        for i in range(5):
+            bound_target_fn = partial(target_fn, idx=i)
+            return_type = self.mp.build_actor(bound_target_fn, builder)
+            # build_actor returns 0 if it is a child process
+            self.assertEqual(return_type, 0)
+
+        self.test_get_actor_pid()
+
+    @unittest.skip
+    def test_multiprocessing_shutdown(self):
+        """
+        Spawns an actor and sends a stop signal.
+        Checks that actor is stopped successfully.
+        """
+        self.test_multiprocessing_spawn()
+
+        actor_list = self.mp.actor_pids
+        self.mp.stop()
+
+        # Check that all actor PIDs no longer exist
+        for actor_pid in actor_list:
+            self.assertFalse(psutil.pid_exists(actor_pid))
+
+    @unittest.skip("Test case for reassigning actors to Processes not yet \
+        implemented")
+    def test_multiprocessing_kill(self):
+        """
+        Spawns an actor and kills it after a certain time.
+        Checks that multiprocessing daemon will automatically restart crashed
+        actor and reassign to correct process.
+        """
+        self.test_multiprocessing_spawn()
+
+        # Gets list of actor PIDs and kills the 1st one in the list
+        actor_list = self.mp.actor_pids
+        os.kill(actor_list[0])
```

Comment:
we kill the actor not from python side.

Reply:
for unit testing, shouldn't we simulate a scenario where the actor "randomly" crashes / dies, to see if the module can automatically restart the actor?

Reply:
for actor "randomly" crashes/dies, I move the restart work to actor in the latest commit. The multiprocessor will send a signal to actor for stopping slowly

Reply:
Can you share with me which file has been changed? In the latest commit (https://github.com/hongdami/lava/commit/b71b5cf8f0a211c88478186e392aedb881bae10e) it seems that only the PyWrapper has been changed